src/sonic-ctrmgrd/ctrmgr/readiness_probe.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+# This script is used by k8s to check the readiness of containers
+# Check if the container is readiness or not, exit code 0 means readiness, others mean not readiness
+
+#### exit code contract, k8s only cares zero or not none-zero, but we want to use none-zero code to indicate different error
+# 0: readiness
+# 1: python script crach exit code
+# 2: supervisor start service doesn't exit normally
+# other exit code: returned by post_check_script, define in the post_check_script, should not include 1,2
+
+# check if the start service exists
+# if the start service exists, check if it exits normally
+# if the start service doesn't exist normally, exit with code 2
+pre_check_service_name="start"
```

Comment:
>start [](http://example.com/codeflow?start=24&length=5)

Will you check all the critical processes?

Reply:
The critical processes unexpected event will be handled by the supervisord exit-listener for now, the listener will kill the container, I don't think we need to check them here. Is this correct?